ygot/render_test.go

```diff
@@ -1580,11 +1580,17 @@ type testAnnotation struct {
 }
 
 func (t *testAnnotation) MarshalJSON() ([]byte, error) {
-	return json.Marshal(*t)
+	t2 := *t
+	t2.AnnotationFieldOne += t2.AnnotationFieldOne
```

Comment:
Can you explain why this is required? It seems to be setting `t2` to itself, do you mean to be setting it to a field of `t`?

Reply:
Is there a reason that there are no test changes associated here? Can we add a test case that covers the bug that reported this at least?

Reply:
Comment added -- this change exposes the bug by deviating the unmarshal/marshal behaviour from `json.Marshal/json.Unmarshal`. This deviation causes the previous code to fail due to its unbalanced usage of using the custom `MarshalJSON` followed by the built-in `json.Unmarshal`.

Reply:
Sorry -- this still doesn't make sense to me. What is the "deviation" that you're referring to here? 

Please could we have a more explicit comment of the form "we change the value of t2 explicitly to ensure that <something doesn't happen>". I think I'm missing context as to why we need the behaviour of this `MarshalJSON` to be different to the behaviour of the builtin `json.Marshal`.

Thanks -- it's very possible I'm just missing something obvious here.

Reply:
Ok updated, the point of this is to prevent the old way of copying -- i.e. `json.Marshal`, then `json.Unmarshal` from succeeding. The previous way was only using the `Annotation` interface for the marshal call, so if the marshal call deviated, `json.Unmarshal` would fail.

Reply:
Thanks -- this is clearer now.